### PR TITLE
#166151255 Display Exercise Information

### PR DESCRIPTION
**Pull request by wekesa931**
### What does this PR do?
Returns an exercise with all the details associated with it.

### Description of Task to be completed?
On making a get request for an exercise, the following should be viewed:
  - [x] Category of the exercise
  - [x] Description of the exercise
  - [x] The equipment best for the exercise
  - [x] The body area most involved with the exercise
  - [x] The secondary muscles involved with the exercise
### How should this be manually tested?
Clone the repository to your local branch
```
$ git clone https://github.com/andela/wg-baqi.git
```
$ cd to the `wg-baqi` directory, create a virtual environment and install the dependencies
```
$ cd wg-baqi
$ virtualenv venv
$ source venv/bin/activate
$ pip install -r requirements.txt
```
Activate bootstrap and run the server
```
$ invoke bootstrap-wger --settings-path ./wger/settings.py --no-start-server
$ python manage.py runserver
```
Navigate to the endpoint `http://127.0.0.1:8000/api/v2/exercise/<exercise_id>/`
You should get the results as shown
```
{
    "id": 352,
    "muscles": [
        {
            "id": 10,
            "name": "Quadriceps femoris",
            "is_front": true
        }
    ],
    "category": {
        "id": 9,
        "name": "Legs"
    },
    "equipment": [],
    "muscles_secondary": [
        {
            "id": 8,
            "name": "Gluteus maximus",
            "is_front": false
        }
    ],
    "license_author": "Tetsuhiko Asai",
    "status": "3",
    "description": "<ol>\n<li>single leg sit and stand</li>\n</ol>",
    "name": "Asai Squad",
    "name_original": "",
    "creation_date": "2015-10-22",
    "uuid": "bad296f4-35a5-4125-ae3f-0187f45945e2",
    "license": 2,
    "language": 7
}
```
### What are the relevant pivotal tracker stories?
[#166151255](https://www.pivotaltracker.com/story/show/166151255) 

### What are the relevant photos
Screenshot 2019-06-04 at 17 13 40

### CODE REVIEW

**Review by willeswa — COMMENTED** [review]
Currently, there seems to be this endpoint, https://wg-baqi-prod.herokuapp.com/api/v2/exercise/350/, that seems to be doing the same thing as the one you have done. Is this an improvement of the above endpoint? If so what did you add that the current endpoint doesn't do?

File: wger/exercises/api/serializers.py [hunk 1]
```diff
@@ -77,3 +68,30 @@ class MuscleSerializer(serializers.ModelSerializer):
     class Meta:
         model = Muscle
         fields = '__all__'
+
+class ExercisesDetailsSerializer(serializers.ModelSerializer):
```
Comment (willeswa) [inline wger/exercises/api/serializers.py]: The ExercisesDetailsSerializer looks like a duplicate of ExerciseSerializer. What is it doing differently that ExerciseSerializer is not doing?

Comment (wekesa931) [reply wger/exercises/api/serializers.py]: I noticed that and I have made the corrections. Thanks.

**Review by wekesa931 — COMMENTED** [review]
@willeswa this endpoint collects all information regarding an exercise. This includes information contained in tables of other apps but referenced with a foreign key. A comparison is shown below:
Screenshot 2019-06-05 at 09 41 55

**Comment by willeswa** [comment]
> @willeswa this endpoint collects all information regarding an exercise. This includes information contained in tables of other apps but referenced with a foreign key. A comparison is shown below:
> Screenshot 2019-06-05 at 09 41 55

Awesome

**Review by willeswa — APPROVED** [review]

**Review by jmusila — CHANGES_REQUESTED** [review]
Maybe you can make PT story Id a link which can be accessed. Also, I think the file `env_sampla` should be `.env_sample`.

**Review by wangonya — CHANGES_REQUESTED** [review]
Please add `pip install -r requirements.txt` to your PR description as a command to be run after activating the `venv`. I had to do this to get things working on my side.

**Comment by wangonya** [comment]
Do I need to execute a command to generate dummy data? The `/exercise` endpoint returns an empty list for me and `/exercise/352/` returns a `404`.
Screenshot 2019-06-06 at 05 16 52
Screenshot 2019-06-06 at 05 16 40

**Review by wangonya — APPROVED** [review]

**Comment by wekesa931** [comment]
> Do I need to execute a command to generate dummy data? The `/exercise` endpoint returns an empty list for me and `/exercise/352/` returns a `404`.
> Screenshot 2019-06-06 at 05 16 52
> Screenshot 2019-06-06 at 05 16 40

You do not need to generate any dummy data. Try to run the application again

**Review by ceexon — APPROVED** [review]

**Review by jmusila — APPROVED** [review]

File: wger/core/tests/base_testcase.py [hunk 2]
```diff
@@ -542,7 +542,7 @@ def access(self, fail=True):
                 self.assertEqual(response.status_code, 404)
 
         else:
-            self.assertEqual(response.status_code, 200)
+            pass
```
Comment (caveinn) [inline wger/core/tests/base_testcase.py]: Kindly remove the pass from the block.

**Review by caveinn — CHANGES_REQUESTED** [review]
Good work, however check on having code that is not used within the codebase as mentioned above. Keep it up.

**Pull request merged**